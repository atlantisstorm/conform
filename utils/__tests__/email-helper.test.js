const EmailHelper = require('../email-helper');

describe('EmailHelper tests', () => {
  it('should build with correct params', () => {
    const body = "{ \"email\": \"test@test.com\", \"name\": \"John Doe\", \"message\": \"This is a very short test message!\" }";
    const myEmail = "john@doe.com";
    const myDomain = "test.com";
    const myDomainTitle = "This is my domain";

    const params = EmailHelper.buildEmailParams(body, myEmail, myDomain, myDomainTitle);
    expect(params).toMatchSnapshot();
  });

  it('should throw error if account details are not supplied', () => {
    const body = "{ \"email\": \"test@test.com\", \"name\": \"John Doe\", \"message\": \"This is a very short test message!\" }";
    const myEmail = "";
    const myDomain = "";
    const myDomainTitle = "";

    expect(() => {
      EmailHelper.buildEmailParams(body, myEmail, myDomain, myDomainTitle)
    }).toThrow("Missing parameters account details, you must supply 'myEmail', 'myDomain', 'myDomainTitle'.");
  });

  it('should throw error if body does not contain expected parameters', () => {
    const body = "{}";
    const myEmail = "john@doe.com";
    const myDomain = "test.com";
    const myDomainTitle = "This is my domain";

    expect(() => {
      EmailHelper.buildEmailParams(body, myEmail, myDomain, myDomainTitle)
    }).toThrow("Missing parameters from body, make sure to add parameters 'email', 'name', 'message'.");
  });  
});