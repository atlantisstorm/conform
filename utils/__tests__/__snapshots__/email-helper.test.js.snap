// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EmailHelper tests should build with correct params 1`] = `
Object {
  "Destination": Object {
    "ToAddresses": Array [
      "john@doe.com",
    ],
  },
  "Message": Object {
    "Body": Object {
      "Html": Object {
        "Charset": "UTF-8",
        "Data": "
  <table style=\\"width: 640px;\\">
    <tbody>
      <tr>
        <td>
          <p style=\\"background-color: #ccc; margin: 0px; padding-top: 5px; padding-bottom: 5px; padding-left: 10px; padding-right: 10px;\\">Message sent today from <b>This is my domain</b> contact page.</p>
          <table style=\\"width: 100%;\\">
            <tbody>
              <tr>
                <td style=\\"background-color: #ddd; margin-top: 5px; margin-bottom: 5px; margin-left: 5px; margin-right: 5px; padding-top: 5px; padding-bottom: 5px; padding-left: 10px; padding-right: 10px;;\\">Name</td>
                <td style=\\"background-color: #eee; margin-top: 5px; margin-bottom: 5px; margin-left: 5px; margin-right: 5px; padding-top: 5px; padding-bottom: 5px; padding-left: 10px; padding-right: 10px;;\\">John Doe</td>
              </tr>
              <tr>
                <td style=\\"background-color: #ddd; margin-top: 5px; margin-bottom: 5px; margin-left: 5px; margin-right: 5px; padding-top: 5px; padding-bottom: 5px; padding-left: 10px; padding-right: 10px;;\\">Email</td>
                <td style=\\"background-color: #eee; margin-top: 5px; margin-bottom: 5px; margin-left: 5px; margin-right: 5px; padding-top: 5px; padding-bottom: 5px; padding-left: 10px; padding-right: 10px;;\\"><a href=\\"mailto:test@test.com\\" style=\\"color: #ff3300;\\">test@test.com</a></td>
              </tr>
              <tr>
                <td style=\\"background-color: #ddd; margin-top: 5px; margin-bottom: 5px; margin-left: 5px; margin-right: 5px; padding-top: 5px; padding-bottom: 5px; padding-left: 10px; padding-right: 10px;;\\" colspan=\\"2\\">Message</td>
              </tr>
              <tr>
                <td style=\\"background-color: #eee; margin-top: 5px; margin-bottom: 5px; margin-left: 5px; margin-right: 5px; padding-top: 5px; padding-bottom: 5px; padding-left: 10px; padding-right: 10px;;\\" colspan=\\"2\\">This is a very short test message!</td>
              </tr>
            </tbody>
          </table>
          <p style=\\"background-color: #ccc; margin: 0px; padding-top: 5px; padding-bottom: 5px; padding-left: 10px; padding-right: 10px;\\">Message sent from <a href=\\"http://test.com\\" style=\\"color: #ff3300;\\" title=\\"This is my domain\\">test.com</a></p>
        </td>
      </tr>
    </tbody>
  </table>",
      },
    },
    "Subject": Object {
      "Charset": "UTF-8",
      "Data": "Message from This is my domain!",
    },
  },
  "ReplyToAddresses": Array [
    "test@test.com",
  ],
  "Source": "john@doe.com",
}
`;
